### PR TITLE
User Management 

### PR DESCRIPTION
**Pull request by Oneill38**
Functionality for adding in new admins/deleting them/promoting them to superadmins. User route is namespaced with admin to avoid devise route collisions. Thoughts on namespacing the dashboard under admin? Might make sense. 

Went with string based roles as opposed to a db table for now, based on https://github.com/CanCanCommunity/cancancan/wiki/Role-Based-Authorization

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		@user.destroy
+		render json: { message: "removed" }, status: :ok
+	end
+
+	def update
+		@user = User.find(params[:id])
+		if @user.update(user_params)
+			render json: { message: "updated" }, status: :ok
+		else
+			render json: { message: @user.errors.full_messages.to_sentence }
```
Comment (cgalarza) [inline app/controllers/admin/users_controller.rb]: Should status also be set here?

File: app/views/admin/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<div>
+	<h2>All Users</h2>
+	<ul>
+		<% @users.each do |user| %>
+			<li>
+				<%= user.email %>
+				<% if user.role == "admin" %>
```
Comment (cgalarza) [inline app/views/admin/users/index.html.erb]: Helpers for these types of checks would make the code a bit cleaner. Something like `user.admin?`

File: app/views/admin/users/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+					<%= link_to "<i class='fa fa-user-circle' aria-hidden='true'></i>".html_safe, admin_user_path(user, user: {id: user.id, role: "superadmin"}), method: :put, remote: :true %>
+				<% elsif user.role == "superadmin" %>
```
Comment (cgalarza) [inline app/views/admin/users/index.html.erb]: See comment above

**Comment by Oneill38** [comment]
Had to change the column default to an empty string since you can't rollback that type of migration

File: spec/support/login.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
```
Comment (cgalarza) [inline spec/support/login.rb:1]: Don't think you need this require in support files.

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<nav class="navbar navbar-default navbar-static-top navbar-inverse">
+	<ul class="user-login">
+    	<% if @current_user && @current_user.role == ("admin" || "superadmin") %>
```
Comment (cgalarza) [inline app/views/layouts/_header.html.erb:3]: Could use helper methods here? It's cool if you want to skip this until we get to working on the roles some more.

**Comment by cgalarza** [comment]
Looks great! Just commented on a few more cleanup things. Thanks for implementing the things we talked about. 
Note that more work on users roles will be done in subsequent pull requests.

**Pull request merged**